src/interp/binary-reader-interp.cc

```diff
@@ -1100,15 +1192,19 @@ Result BinaryReaderInterp::OnCallIndirectExpr(Index sig_index,
 Result BinaryReaderInterp::OnReturnCallExpr(Index func_index) {
   FuncType& func_type = func_types_[func_index];
 
-  Index drop_count, keep_count;
+  Index drop_count, keep_count, catch_drop_count;
   CHECK_RESULT(
       GetReturnCallDropKeepCount(func_type, 0, &drop_count, &keep_count));
+  CHECK_RESULT(
+      validator_.GetCatchDepth(label_stack_.size() - 1, &catch_drop_count));
   // The validator must be run after we get the drop/keep counts, since it
   // will change the type stack.
   CHECK_RESULT(validator_.OnReturnCall(loc, Var(func_index)));
   istream_.EmitDropKeep(drop_count, keep_count);
+  istream_.EmitCatchDrop(catch_drop_count);
 
   if (func_index >= num_func_imports()) {
+    istream_.Emit(Opcode::InterpAdjustFrameForReturnCall, func_index);
```

Comment:
Looking at the code, I do notice that `GetLocalType` looks up the function reference from the stack frame, so it might be possible to construct a failing case for other programs too. I'll look into this (will file a separate issue/PR if there's a bug).

Reply:
So it's hard to produce a failing test case for normal operation, the frame behavior of `return_call` did break the tracing in the interpreter. For example:

```
$ wabt/out/clang/Debug/wasm-interp out/test/interp/return-call-get-local/return-call-get-local.wasm -t --run-all-exports --enable-tail-call
>>> running export "f":
#0.    0: V:0  | i64.const 1
#0.   12: V:1  | br @24
#0.   24: V:1  | i32.const 3
#0.   32: V:2  | drop_keep $1 $1
#0.   44: V:1  | br @68
#0.   68: V:1  | i32.const 2
wasm-interp: ../../../src/interp/interp.cc:192: wabt::interp::ValueType wabt::interp::FuncDesc::GetLocalType(wabt::interp::Index) const: Assertion `iter != locals.end()' failed.
Aborted (core dumped)
```

when running the example in issue #1761.

The change in this PR doesn't quite fix this bug (in addition to updating the function reference, the value stack height needs to be updated) but it could be easily done in a follow-up.

Reply:
I didn't mean to ask you to investigate an existing bug, but thanks for the fix!